### DISCUSSION TITLE
Backward vehicles as a option in OpenRCT2

### DISCUSSION THREAD

**Discussion by TheLarry98** [OpenRCT2/OpenRCT2#15701] [Feature requests] [answered]
In RCT1 and RCT2 we had some trains which could run backwards. The problem was that there were only a few which could that and those were the exact same vehicles only backwards. We also not had every backward vehicle in RCT2 which we had in RCT1. So it would be much better when there would be a option for every roller coaster.

Rollercoasters which had the feature in RCT1 and RCT2
*Steel Roller Coaster (Looping Roller Coaster in RCT2) (only in RCT1)
*Wooden Roller Coaster (only in RCT1)
*Wooden Roller Coaster (6 seater) (only in RCT2)
*Heartline Twister Roller Coaster (Heartline Twister Coaster in RCT2)
![b9538952957741118875c53183d59ed3](https://cloud.githubusercontent.com/assets/12634026/19898732/8b4932ae-a05d-11e6-9c45-9407520e637d.jpeg)
#

**Comment by SilvariusCoyote** [comment]
This would require separate .DATs for each backward vehicle. I think it would be a better idea to ask some ridemakers (such as @X123M3-256 ) to make custom .DATs with reversed sprites. In fact, you could probably make a program which takes a .DAT and reorders the sprites to make a backwards train .DAT.

**Comment by jensj12** [comment]
Since all sprites are already in the file at known locations, wouldn't it be possible to add a 'reverse' checkbox to reverse any type of train? For a certain sprite, just make it point at the reversed sprite instead. I have no idea how special cars at both ends of the train are handled so I'm not sure if reversing the positions of those is possible.

edit: I checked the code (removed some duplicate code while doing so), and it's quite trivial to reverse the order of the cars with a checkbox, but only the order. Drawing the backwards sprite is more complicated, and so is making powered rides work.

**Comment by TheLarry98** [comment]
@jensj12 This is exactly what I thought! Otherwise every vehicle would be double in the game.

**Comment by SilvariusCoyote** [comment]
@jensj12 that's a good idea, but I don't think there's any space in the .sv6 file structure for a reverse variable.

**Comment by X123M3-256** [comment]
Technically I don't think it'd be too hard to implement, but like @YoloSweggLord said, this would almost certainly require a new file format or at least some alterations to the current one if you want it to persist across savegames. The workaround is just to use a seperate .DAT with the sprites suitably rearranged (in fact, this is exactly how reversible trains are implemented on the wooden coaster). Yes, it occupies two object slots, but in most cases you'd be unlikely to have both the reversed and non-reversed versions loaded at once (exception is with the wooden coaster since it is/was common to have a racing coaster with one side forward and one reversed).

@jensj12 has a good point though - the current object format is not flexible enough to reverse arbitrary trains like this because you can specify the second car in the train independently and not the second to last. Now, you could work around _this_ by reversing the sprites but not their position in the train, and then rearranging the vehicle order in game. However, this last step can be a bit awkward, particularly if you aren't building the game from source.

**Comment by SilvariusCoyote** [comment]
I understand that RCT1 uses different mechanics and filetypes, but couldn't we investigate the reversed looping coaster train .DAT (or whatever RCT1 uses) and see if its method of making the last car special can or can't be applied to RCT2?

**Comment by X123M3-256** [comment]
RCT1 does not have .DAT files, everything is hardcoded. And even if it did, and there was a reverse flag, RCT2 does not have it.

Besides, you can create a reversed looping coaster train DAT, it's things like the steam train where the _second_ car is different that you can't. And even then, the restriction is only in the DAT - once loaded into the game you can have any train makeup you want, but like I say, this is awkward because OpenRCT2 doesn't have a cheat to do it, so you have to tweak the SV6 manually.

**Comment by SilvariusCoyote** [comment]
Oh, I misunderstood your first comment. And thanks for correcting me on RCT1's functionality, I don't have RCT1 so I don't know about it's behavior.

**Comment by ZeeMaji** [comment, marked as answer]
Implemented in #19305.
